test/commands/extended/is_promotable_test.py

```diff
@@ -0,0 +1,407 @@
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '9999999999999999999999999999999999999999999999999999999999999999'
+           '99999999999ETTEXDKDEUALTLRJVXRHCPRJDLGPJCEQBJOMOAGBZWZCWLNUEWHAU'
+           'SYJMYPEZPYNBTPSPGUIPQ9VOUNQ9999999999999999999999999999JVPROMOTA'
+           'BLETEST99999999999USHRPBD99999999999999999999XFVLEXEJPTYI9TUA9UL'
+           'FNHXBGDUCOEPDIBKSZFXEBO9HFEGLENBCOVKHZ99IWZVCVSTUGKTIBEOVFBJPCDA'
+           'WCMHRLDQPBBGISNENMIXOGGYSRYXGAFEJC9FOLXLYIQVUHFCMVRPBIEAXDUYYPYN'
+           'EZPHH9KB9HZ9999DAWCMHRLDQPBBGISNENMIXOGGYSRYXGAFEJC9FOLXLYIQVUHF'
+           'CMVRPBIEAXDUYYPYNEZPHH9KB9HZ9999ISPROMOTABLETEST99999999999IOCFQ'
+           'OIOF999999999MMMMMMMMMCAA9999999UYA99999999999999'
+        )
+
+        self.hash2 = TransactionHash(
+            'FLNPRAOEYMBIXZBBFMQGCEWLRKTZTMWWTVUQRNUNMZR9EMVKETRMWHRMBFWHJHX'
+            'ZOIMUWZALX9IVZ9999'
+        )
+
+    def test_wireup(self):
+        """
+        Verify that the command is wired up correctly.
+        """
+        self.assertIsInstance(
+            Iota(self.adapter).isPromotable,
+            IsPromotableCommand,
+        )
+
+    def test_happy_path(self):
+        """
+        Successfully checking promotability.
+        """
+        # To pass timestamp check, we need a timestamp that defines
+        # a moment no earlier than 660s from the moment the function
+        # is called during test. To do this, we cheat a bit:
+        tx = Transaction.from_tryte_string(self.trytes1)
+        tx.attachment_timestamp = get_current_ms()
```

Comment:
❔ Should we mock `get_current_ms` so that we can more easily control the outcome?

Reply:
in this method yes. But in `test_one_timestamp_invalid` it causes more headache, because then the 'modified' transaction becomes invalid and will not have a valid hash that can be compared to `self.hash2`